### ISSUE TITLE
bug with `property-value-blacklist`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#972]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Strange error

> Which rule, if any, is this issue related to?

e.g. `property-value-blacklist`

> What CSS is needed to reproduce this issue?

e.g.

```css
a {
    font-size: 1em;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "property-value-blacklist": {}
  }
}
```

```json
{
  "rules": {
    "property-value-blacklist": {
      "position": ["fixed"]
    }
  }
}
```

> Which version of stylelint are you using?

e.g. `5.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS, CSS

> What did you expect to happen?

Nothing

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
TypeError: Cannot read property '0' of undefined
    at testAgainstString (/usr/lib/node_modules/stylelint/dist/utils/matchesStringOrRegExp.js:23:39)
    at testAgainstStringOrArray (/usr/lib/node_modules/stylelint/dist/utils/matchesStringOrRegExp.js:19:12)
    at exports.default (/usr/lib/node_modules/stylelint/dist/utils/matchesStringOrRegExp.js:11:10)
    at /usr/lib/node_modules/stylelint/dist/rules/property-value-blacklist/index.js:27:44
    at /usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:86:28
    at /usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:73:26
    at Rule.each (/usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:60:22)
    at Rule.walk (/usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:72:21)
    at /usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:75:32
    at Root.each (/usr/lib/node_modules/stylelint/node_modules/postcss/lib/container.js:60:22)
```

**Comment by davidtheclark** [comment]
This was a pretty important bug to uncover. Thanks.
